### DISCUSSION TITLE
Quadratic Staking

### DISCUSSION THREAD

**Discussion by midenaio** [idena-network/idena-docs#69] [IIPs]
Idena core team has been receiving many requests from the community to add percentage yield to the identity stake and allow stake replenishment. Staking can make the iDNA coin more attractive for investors and reduce inflation by locking coins in stakes. 

However, the traditional (linear) staking model leads to inequality because large coin holders can extract the major part of the profits from staking. With Proof-of-Personhood in mind, we can offer [quadratic staking reward schemes](https://docs.idena.io/docs/iip/iip-4) that are free of capital bias. We invite the community to consider this quadratic staking proposal and discuss its advantages:
- Quadratic staking can make the iDNA coin more attractive for investors and reduce inflation by locking coins in stakes. This can lead to faster network growth.
- Quadratic staking can make an older identity more valuable and would incentivize people to not terminate their identities.
- Quadratic staking can improve security of the protocol because larger stakes increase the value of each identity. In other words, quadratic staking will lead to the growth of the capitalization of many identities resulting in higher security of the network.

Identity’s stake is a security measure that ensures that the private key of a valid identity cannot be purchased without the risk of losing the stake and identity status. We see benefits in being able to replenish the identity stake and get rewarded for that but only if it leads to a balanced rise in stakes for a large number of identities. We consider the staking to be undesirable when it leads to the domination of a few large coin holders. We propose quadratic staking that discriminates against large coin holders and encourages smaller players to increase their stakes.

**Comment by simontakouridis** [comment]
A very interesting IIP, overall I agree with it, but there are some serious concerns.

1. If you lose your human status, your stake is at serious risk of being lost. This means people will terminate their identity, which goes against the whole purpose of the iip, which is to incentivise people not to terminate - a little silly really. A few ideas to fix this issue are - 1) allow suspended status to keep their stake if it fails, 2) have two validation ceremonies per epoch, that way people can maintain a couple of identities that can serve as a "backup" in case one needs to be terminated. 3) Have a mechanism to slowly withdraw your stake, without having to terminate your identity.

2. The "quadratic" rate of 0.9 is very high, and disproportionately favours whales (who already hold most of the circulating  idna). The justification for this was to prevent pools from getting too powerful, however imo pools play a very important role in idna, and are highly incentivised to expand the network size. I think 0.8 or 0.85 would be more reasonable.

3. I imagine most "investors" would use a hardware wallet to secure their private keys. If they want to stake, then they would need to transfer their funds to an account where the private key is "exposed" to the computer that is running the Idena client - which is a security risk. I propose an update to the client that allows users to connect their hardware wallet to the client, instead of storing the private key on the computer.

Great IIP, but needs some changes as recommended.

**Reply by ubiubi18** [reply]
What about allowing to delegate another wallets (for example your separate cold wallets) balance to your main identity stake? Delegated staking? You can only undelegate that part of the stake again if the main identity is terminated, no other way to undelegate. But in case the main identity is terminated by flip session failure (or by hack attack or theft), your separate cold wallets stake is just undelegated again and stays where it is under control of a separate private key again - instead of beeing burned or stolen? Out of reach of hackers, because it has its own private key, also out of reach of protocol burning?

**Comment by reb0rn21** [comment]
I agree with the change buy human status need more protection, I still hate last 10 epoch score (update) and still think its the reason if more then few bad flips hit you, there is chance for score to drop huge and high level identity will chose to kill himself (I already did that no way I am gona risk money on some bad flips out of my control) 
With more stake this is more problem to score risk!

ppl need security if we want for them to stake they for sure will not be happy to risk it

**Comment by Tirion77** [comment]
The recent update to this proposal created some interesting discussion in the community. Namely, I mean this commit:

"The proposal extends existing mechanics of the stake protection. Identities with `Verified`, `Suspended` or `Zombie` status having age above or equal `5` epochs will have stake protection. It means that their stake will be transferred to the normal wallet if the validation is failed or missed."

Let's revisit the goals and advantages of this staking mechanism:

(1) make the iDNA coin more attractive for investors via reduction in circulation and introduction of a yield on the stake
(2a) make older identities more valuable 
(2b) and disincentivize killing of identities for the stake
(3) increase value of identities by tying the new yield (thus, additional income) to identities

And now review them in light of this new potential change with extension of staking protection. 

The current burn mechanics could potentially ward off investors from providing a large stake, as the whole stake burns in the event of a failure for 'Verified', 'Suspended', or 'Zombie', and goal 1 would not be fully satisfied. So, it makes sense to introduce additional stake protection against the burn, removing investor worries and helping goal 1.

This doesn't change much for goal 3, so I won't comment there.

However, while this also succeeds in disincentivizing killing of identities (2b) because you'll now always get your stake even during failures, it doesn't quite achieve goal 2a unless the goal is to just reach a certain point (age >5). It seems that the best strategy would be to get an identity to age 5, stake, invite another identity for yourself, attempt multiple validations, get it to age 5, stake etc. This would both increase your potential income (due to the quadratic nature of the staking rewards distribution) and have the same benefits for every identity of age >5 with no downside at all -- you get your stake back anyway! Older identities do not get valuable. In fact, the increase of an identity's value based on age flattens after age 5, and, even worse, it encourages attempts of validating with more identities in search of better yields by removing the stake burn fears.

I propose a change to the extension in stake protection that seeks to compromise on reducing the risk for investors while maintaining the value of older identities over several small ones:
1. Introduce a minimum amount of stake burnt in the event of a failure. Perhaps 10%, a number in line with the power of 0.9 used for staking calculations. This is to always have an incentive to validate correctly.
2. Introduce additional burn penalties for younger identities through a sliding scale based on an identity's age. This scale would depend on the oldest identity participating in that validation session, with the oldest identity having 0% additional burn penalty. A possible scale could be defined by a square root of the ratio between the identity's age and the oldest identity -- **sqrt(identity_age/oldest_identity_age)**. 
An example equation to calculate the total possible burn penalty would be **0.1+0.9(1-sqrt(identity_age/oldest_identity_age))**, where 0.9 is used to account for the remaining possible 90% of the penalty. Of course, we could change that to be something else, if we want to cap total possible burn. I'll provide examples below. This way, we create a burn system that both encourages to keep your identity as old as possible, but still lets younger identities get protection quickly. Moreover, it is a square root letting us use the word "quadratic" for this IIP on more grounds :) That said, we could make this a cubic root instead, thus letting younger identities catch up on the penalty reduction even quicker.

For example, in a validation session with the oldest identity being 100, an identity of age 3 could stand to lose:

10% minimum burn
0.9*(1-sqrt(3/100))=0.9*(1-0.17320508075)=0.74411542731=74.41% of penalty burn
total: 84.41%

whereas an identity of age 10 would get a total possible burn of 71.5%.

Example graphs:
1. With burn penalty of up to 90% (i.e. possible to burn the whole stake with the initial minimum burn)
![image](https://user-images.githubusercontent.com/47679936/159340544-1181aa51-da5a-4c95-b733-1cfb90e68cd6.png)
2. With burn penalty of up to 90%, 70% (total possible 80%), and 50% (total possible 60%):
![image](https://user-images.githubusercontent.com/47679936/159340622-387ff102-250e-4549-91ae-e9b10d88a1f9.png)

As a result we:
1. Reduce investor fears by reducing the burn penalties
2. Give a use case to the age, giving older identities real value and discouraging identity kills.
3. Maintain the integrity of incentives to validate correctly, i.e. keep the stake burn fears
4. Get to call this IIP truly quadratic :)

**Reply by turbosandbox** [reply]
Great feedback, I strongly agree with this suggestion. In particular I believe there must be something at stake for it to be continued called "stake" and that a minimum burn for failing. The 5 epoch step up stake immunity is too rewarding.

`Introduce a minimum amount of stake burnt in the event of a failure. Perhaps 10%, a number in line with the power of 0.9 used for staking calculations. This is to always have an incentive to validate correctly.`

**Reply by reb0rn21** [reply]
All ppl that stake wish to validate correctly, but you can not disregard huge number of bad flips in validation which are not filtered by ppl reporting it
If that make ppl scared they will not risk to have possibility to be burn mostly from failed score under 92% which will happen over long period of time when you get hit by bad flips at once

Idena need to be less scare to newbies and old and not more

**Reply by turbosandbox** [reply]
You don't have to add to your stake, the reward payout will remain about the same. The staking system with full 100% burn already exists now.

Your beef is with bad flips, or you want to get rid of stake completely. The former is not related to this topic.

**Comment by turbosandbox** [comment]
I believe it makes more sense to reduce rather than increase overall reward distributions for large pools with many identities by adding the sum of all pool identities' stakes first. Then we can be flexible on choosing the quadratic power `p`.

The distribution of rewards for staking depends on the weight of identity or pool node `n` calculated as the amount of coins locked in stake to the power of `p`.

```
total_node_stake[n] = sum ( stake[i] for i in n )
weight[n] = total_node_stake[n]^p
W = sum( weight[n] )
share[n] = weight[n]/W
reward[n] = F\W * share[n]
r[n] = reward[n] / stake[n]
```

Then distribute pool reward for the identity `i` in pool node `n` linearly:

```
weight[i] = stake[i]/total_node_stake[n]
r[i] = weight[i] * r[n]
```

Alternatively, the pool `n` reward could also distributed across identities using the quadratic weighing system to encourage distribution of stake within a pool.

This also has the added benefits of encouraging users to run their own nodes increasing node decentralization.

**Comment by midenaio** [comment]
@Tirion77 @reb0rn21 @simontakouridis @nudesushi @ubiubi18 thanks for your feedback.

We've tried to address concerns about the stake protection (see [changes](https://github.com/idena-network/idena-docs/pull/72/files)).

**Reply by Zen-44** [reply]
Is it possible to clarify if the proposed stake protection scheme affects the entire stake (20% from mining, validation + the amount someone locks up) or just the coins voluntarily locked up?

**Reply by Zen-44** [reply]
Another thing that I think needs clarification is if the locked coins (60% discrimination for new identities) of newbie identities are taken into consideration for their rewards while they sit in the stake. Ideally, rewards should be given only for coins that can't be used ever again unless the identity is killed, at least in my vision.

**Reply by simontakouridis** [reply]
Unfortunately I do not believe the stake protection measures you propose are sufficient. The thesis behind this staking proposal is to 1) make Idena more attractive to investors, and 2) to incentivise users to not terminate their identity - both of which it fails. Because 1) new investors will be at risk of losing some or all of their invested stake for the first 10 epochs, and 2) investors must terminate their identity to have their stake investment returned along with any rewards.

Fortunately there is a solution to the above as discussed on Discord and briefly repeated below:
1) Having a vesting schedule for any added stake e.g. 12 epochs, after which it is then automatically returned to the wallet (along with the rewards).
2) Stake protection is guaranteed, though if withdrawing stake from within the vesting period, staking rewards are forfeited (termination no longer necessary to access stake).

This solution provides adequate investor protection because the added stake is not at risk (only the stake reward), and also incentivises identity longevity.

**Reply by dalikexterminate** [reply]
1. complex mathematics.  maybe instead of inventing 5% 4% and so on, it is much easier to introduce the second type of staking.  investment stake.  which does not burn out under any conditions, but simply goes to the main wallet.  Something like that.  otherwise, no one will stake when there is a risk of losing money due to a protocol error.  such thoughts.  otherwise it's a roll.  2. I saw a discussion of pools here.  So in the pools, the steak lies on more than one account.  the main payouts fall on one account, and the stake is distributed to each on his own.  this mechanism is only for investors (or for pools as investors already AFTER the withdrawal of regular earnings (not a stake)) . That is, I, as the owner of the pool, will receive a smaller share of inflation.  But if this leads to an increase in the rate, then it will be more profitable for me and my 1000+ workers in the end. and will contribute to the even greater growth of the network. I think we can try.

**Comment by reb0rn21** [comment]
its a step in right direction, forcing ppl to keep old identity on life to gain more protection.... 
I can not talk for the ones that are <10 epoch at least burn % is not big so for some not soo huge stake it might ok

also with more risk of low score due 10 epoch score we have I still do not see point of it, the number of possible bad flips to pass the filter is still quite huge and in 2-3 epoch with very bad luck you have big chance to drop under 92%

yeah yeah some say this is not the right topic, but it is every added risk will put ppl of staking!
